homeassistant/components/voip/assist_satellite.py

```diff
@@ -0,0 +1,165 @@
+"""Assist satellite entity for VoIP integration."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncIterable
+import logging
+import time
+from typing import TYPE_CHECKING, Final
+from uuid import uuid4
+
+from homeassistant.components import stt
+from homeassistant.components.assist_pipeline import (
+    PipelineEvent,
+    PipelineEventCallback,
+    PipelineEventType,
+    PipelineStage,
+    async_pipeline_from_audio_stream,
```

Comment:
Only import things for typing. All function calls on the assist pipeline integration need to be done inside the assist satellite base entity class.